packages/cpp/tools/Dockerfile.worker

```diff
@@ -0,0 +1,104 @@
+# Start with the latest Alpine base image for the build stage
+FROM alpine:3.14 AS builder
+ARG GRPC_VERSION=v1.54.0
+
+# Install all the necessary dependencies required for the build process
+# These include tools and libraries for building and compiling the source code
+RUN apk update && apk add --no-cache \
+    git \
+    gcc \
+    g++ \
+    build-base \
+    autoconf \
+    automake \
+    libtool \
+    curl \
+    c-ares \
+    c-ares-dev \
+    make \
+    cmake \
+    unzip \
+    linux-headers
+
+# Clone the gRPC repository using the specified version
+WORKDIR /tmp
+RUN git clone -b $GRPC_VERSION https://github.com/grpc/grpc
+
+# Build the gRPC libraries and binaries
+WORKDIR /tmp/grpc
+RUN git submodule update --init
+RUN mkdir -p cmake/build && \
+    cd cmake/build && \
+    cmake -DCMAKE_INSTALL_PREFIX=/app/install \
+    -DgRPC_INSTALL=ON \
+    -DCMAKE_BUILD_TYPE=Release \
+    ../.. && \
+    make -j $(nproc) && \
+    make install && \
+    ln -sf /app/install/lib/libgrpc++.so /app/install/lib/libgrpc++.so.1 && \
+    ln -sf /app/install/lib/libgrpc++_reflection.so /app/install/lib/libgrpc++_reflection.so.1
+
+# Update the PATH environment variable to include the gRPC libraries and binaries
+ENV PATH="/app/install/lib:$PATH"
+ENV PATH="/app/install/bin:$PATH"
+
+# Remove the gRPC source folder to save space in the final image
+RUN rm -rf /tmp/grpc

```

Comment:
This does not work the way you think. New layers can only *increase* the image size, never decrease.

Reply:
To add more information: RUN, COPY and ADD docker functions are creating new layer. Here you probably want to move this line to line 40 (without the RUN)